docs/DeveloperGuide.md

```diff
@@ -2,37 +2,381 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+This application was developed as part of a university software engineering module CS2113. Some ideas were adapted from:
+1. Duke Java Project
+2. JavaFX Documentation
+3. Official Java 17 Documentation
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+`//include architecture diagram here`
+
+The Architecture Diagram given above explains the high-level design of the HealthBud.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main Components**
+
+`HealthBud` is in charge of the CLI to launch and the exit.
+
+- At CLI launch, it initializes the other components in the correct sequence, and connects them up with each other.
+- At exit, it shuts down the other components and invokes cleanup methods where necessary.
+
+The bulk of the CLI's work id done by the following components:
+
+- [**`UI`**](#ui): Responsible for displaying messages to the user.
+- [**`Parser`**](#parser): Interprets user commands and constructs the corresponding `Command` objects.
+- [**`Command`**](#command): Contains the logic for executing user commands.
+- [**`LogList`**](#loglist): **(New)** Maintains lists for each log type (Meal, Workout, Water, etc.).
+- [**`Storage`**](#storage): Handles reading and writing data to and from the hard disk.
+- [**`Data`**](#data): Holds the in-memory data (logs) of the app.
+
+### UI
+
+![UI_Class_Diagram](images/UiCD.png) <br>
```

Comment:
Would be good to hide the icons as per Prof's recent email, otherwise clear diagram!